src/Qdrant.php

```diff
@@ -5,42 +5,58 @@
  */
 namespace Qdrant;
 
+use Psr\Http\Client\RequestExceptionInterface;
 use Psr\Http\Message\RequestInterface;
 use Qdrant\Endpoints\Cluster;
 use Qdrant\Endpoints\Collections;
 use Qdrant\Endpoints\Service;
 use Qdrant\Endpoints\Snapshots;
-use Qdrant\Http\HttpClientInterface;
+use Qdrant\Http\Transport;
 
 class Qdrant implements ClientInterface
 {
-
-    public function __construct(private HttpClientInterface $client)
+    public function __construct(private readonly Transport $transport)
     {
     }
 
     public function collections(string $collectionName = null): Collections
     {
-        return (new Collections($this->client))->setCollectionName($collectionName);
+        return (new Collections($this))->setCollectionName($collectionName);
     }
 
     public function snapshots(): Snapshots
     {
-        return new Snapshots($this->client);
+        return new Snapshots($this);
     }
 
     public function cluster(): Cluster
     {
-        return new Cluster($this->client);
+        return new Cluster($this);
     }
 
     public function service(): Service
     {
-        return new Service($this->client);
+        return new Service($this);
     }
 
     public function execute(RequestInterface $request): Response
     {
-        return $this->client->execute($request);
+        try {
+            $res = $this->transport->sendRequest($request);
+
+            return new Response($res);
+        } catch (RequestExceptionInterface $e) {
+            var_dump('RequestException', $e->getMessage(), $request); exit();
+            /*$statusCode = $e->getResponse()->getStatusCode();
+            if ($statusCode >= 400 && $statusCode < 500) {
+                $errorResponse = new Response($e->getResponse());
+                throw (new InvalidArgumentException($e->getMessage(), $statusCode))
+                    ->setResponse($errorResponse);
+            } elseif ($statusCode >= 500) {
+                $errorResponse = new Response($e->getResponse());
+                throw (new ServerException($e->getMessage(), $statusCode))
+                    ->setResponse($errorResponse);
+            }*/
+        }
```

Comment:
Exceptions work on Guzzle with specific parameters (`'http_errors' => true`) and are not supported or have different configurations for others. Also, `getResponse()` is not supported by the interface. 😢

Reply:
We need more tests on this part, but according to checking with `symfony/HTTP-client` and `Nyholm/psr7`, I provide some solutions using response->statusCode()